feature-utils/poly-analysis/src/utils/analysis-util.js

```diff
@@ -16,3 +16,28 @@ export async function relevantZipEntries(zipFile) {
             !entry.path.includes("__MACOSX")
     );
 }
+
+export function groupActivitiesByTime(activityDates) {
+    let groupedActivities = { total: 0, values: {} };
+
+    activityDates.forEach((date) => {
+        const activityYear = date.getFullYear();
+        const activityMonth = date.getMonth();
+        if (
+            groupedActivities?.values?.[activityYear]?.values?.[activityMonth]
```

Comment:
Is it possible that the *odd* question marks are not really needed?
1. They're not really checked
2. If they exist, they will have been initialized with `values`, so it's enough to have question marks after `values`.